app/commands/incident.py

```diff
@@ -361,3 +373,171 @@ def generate_success_modal(body):
+
+            for message in messages:
+                # convert the time which is now in epoch time to standard EST Time
+                message_date_time = convert_epoch_to_datetime_est(message["ts"])
+                # get the user name from the message
+                user = client.users_profile_get(user=message["user"])
+                # get the full name of the user so that we include it into the timeline
+                user_full_name = user["profile"]["real_name"]
+
+                # get the current timeline section content
+                content = get_timeline_section(document_id)
+
+                # if the message already exists in the timeline, then don't put it there again
+                if message_date_time not in content:
+                    # append the new message to the content
+                    content += (
+                        f"{message_date_time} {user_full_name}: {message['text']}"
+                    )
+
+                    # if there is an image in the message, then add it to the timeline
+                    if "files" in message:
+                        image = message["files"][0]["url_private"]
+                        content += f"\nImage: {image}"
+
+                    # sort all the message to be in ascending chronological order
+                    sorted_content = rearrange_by_datetime_ascending(content)
+
+                    # replace the content in the file with the new headings
+                    replace_text_between_headings(
+                        document_id, sorted_content, START_HEADING, END_HEADING
+                    )
+        except Exception as e:
+            logger.error(e)
+
+
+# Execute this function when a reaction was removed
+def handle_reaction_removed(client, ack, body, logger):
+    ack()
+    # get the channel id
+    channel_id = body["event"]["item"]["channel"]
+
+    # Get the channel name which requires us to use the conversations_info API call
+    result = client.conversations_info(channel=channel_id)
+    channel_name = result["channel"]["name"]
+
+    if channel_name.startswith("incident-"):
+        try:
+            # Fetch the message that had the reaction removed
+            result = client.conversations_history(
+                channel=channel_id,
+                limit=1,
+                inclusive=True,
+                oldest=body["event"]["item"]["ts"],
+            )
+            # get the messages
+            messages = result["messages"]
+            # if the lenght is 0, then the message is part of a thread, so get the message from the thread
+            if messages.__len__() == 0:
+                # get thread messages
+                result = client.conversations_replies(
+                    channel=channel_id,
+                    ts=body["event"]["item"]["ts"],
+                    inclusive=True,
+                    include_all_metadata=True,
+                )
+                messages = result["messages"]
```

Comment:
Could this code be hoisted out into a shared function since it's the same (I think) as what's happening in the `handle_reaction_added` event handler?

Reply:
Good suggestion - I have made the change.